### ISSUE TITLE
Steam crashes on postmarketOS

### ISSUE THREAD

**Issue by CalcProgrammer1** [Raezroth/Linux-ARM-Gaming-Chroot#15]
I've tried this numerous times on my OnePlus 6T running pmOS and I can get the chroot installed, but I can't get Steam to launch.  It downloads Steam but then when it goes to launch (basically where I'd expect the login prompt to come up), it just closes.  Box86/64 is definitely starting as I see the terminal messages from it.  I haven't tested on pmOS on PinePhone/Pro, but the same setup works fine on Arch on PinePhone Pro.

**Comment by Raezroth** [comment]
Do you have a log you can upload??? 

Launch Steam w/ ::::: BOX86_LOG=1 BOX64_LOG=1 steam +open steam://open/minigameslist -noreactlogin 2> tee -a ~/steam.log >> ~/steam.log

I wish to investigate this further.

**Comment by CalcProgrammer1** [comment]
```
user@oneplus-fajita:/$ BOX86_LOG=1 BOX64_LOG=1 steam +open steam://open/minigameslist -noreactlogin 2> tee -a ~/steam.log >> ~/steam.log
bash: tee: Permission denied
```

**Comment by CalcProgrammer1** [comment]
Without >2 and after:

![image](https://user-images.githubusercontent.com/1051772/215652005-0e1e79b3-55ef-44bd-b5a2-bb8942e3ef9b.png)

**Comment by Raezroth** [comment]
Maybe the permissions for the chroot are borked. tee is a user command. It's suspicious that you don't have permissions to it. Especially since you are in a debian chroot.

**Comment by CalcProgrammer1** [comment]
After updating Steam it died

**Comment by Raezroth** [comment]
anything in the log file??

**Comment by CalcProgrammer1** [comment]
[steam.log](https://github.com/Raezroth/Linux-ARM-Gaming-Chroot/files/10542276/steam.log)

**Comment by Raezroth** [comment]
what verseion of glibc do you have??

**Comment by Raezroth** [comment]
Look for __dn_expand in loaded elfs
Found symbol with version GLIBC_2.4, value = 0xf7f49235
Look for __res_query in loaded elfs
Found symbol with version GLIBC_2.4, value = 0xf7f4e551

**Comment by Raezroth** [comment]
anyway to try with glibc2.35

**Comment by CalcProgrammer1** [comment]
If you're not aware, postmarketOS (Alpine Linux) uses musl instead of glibc.  I wouldn't think this applies in a chroot though, because the chrooted programs should be using the chroot's libc and Debian uses glibc.  I'm not sure what version is installed, whatever is in Debian at this time.

I installed the chroot today using my postmarketos_fixup branch rebased on master (which doesn't change a whole lot, just fixed some install hiccups I had).

**Comment by Raezroth** [comment]
This would fall under the chroot's glibc. Sorry I should have specified

**Comment by CalcProgrammer1** [comment]
ldd --version
ldd (Debian GLIBC 2.36-8) 2.36

**Comment by Raezroth** [comment]
Are you using box86/64 builds built on the device?  I am still puzzled though. May have just be a library issue. Could be solved by a different version of debain as well

**Comment by CalcProgrammer1** [comment]
No, I'm using the versions downloaded as part of the install script.  It is either something with the OnePlus 6T or it's something with postmarketOS.  I installed the chroot on the same day on my OP6T with pmOS and my PinePhone Pro with Arch and the PinePhone Pro worked fine.  I stupidly left my PPP at my parents' house over Christmas break though, so I can't test PPP pmOS now.  I could install it for the OG PP I guess.

**Comment by Raezroth** [comment]
Build box86/64 on the oneplus 6t. I need to change the versions downloaded in the script. I should compile my own again. maybe set up a repo and a debian chroot distro for this. I don't know yet

**Comment by Raezroth** [comment]
Too many issues with the included repos/binaries. Need a better solution now

**Comment by CalcProgrammer1** [comment]
Going to try setting it up as a docker container and see if that works.

**Comment by CalcProgrammer1** [comment]
Made progress using a Docker container

I ran with -noreactlogin but still...

![image](https://user-images.githubusercontent.com/1051772/218830908-1bc37506-12e7-458b-a40c-c47b0913dfc9.png)

**Comment by Raezroth** [comment]
I think they finally removed that from launch options. There is going to
need to be more work done on box86/64 for that login to work. You should
still be able to use Big Picture.

**Comment by CalcProgrammer1** [comment]
Pretty sure old big picture mode is gone now too.

**Comment by CalcProgrammer1** [comment]
I have now successfully run Steam and some games using both FEX and box86/64 in postmarketOS.  I wrote up a guide for FEX on the postmarketOS wiki here:

https://wiki.postmarketos.org/wiki/Steam

I will do another write up on setting up box86/box64.  I did both setups in Distrobox containers, which are easier to manage and set up than plain chroots.  I'm working on a script to install box86/box64/Steam on a fresh Debian Distrobpx container as well.
